src/graphql/resolvers.ts

```diff
@@ -69,12 +73,27 @@ export async function queryMulti(parent, args, context: ResolverContext, info) {
   return result.map(item => formatItem(item, jsonFields));
 }
 
-export async function querySingle(parent, args, context: ResolverContext, info) {
+export async function querySingle(
+  parent,
+  args,
+  context: ResolverContext,
+  info: GraphQLResolveInfo
+) {
   const returnType = getNonNullType(info.returnType) as GraphQLObjectType;
   const jsonFields = getJsonFields(returnType);
 
   const id = parent?.[info.fieldName] || args.id;
 
+  const parsed = parseResolveInfo(info);
+  if (parsed) {
+    // @ts-ignore
+    const simplified = simplifyParsedResolveInfoFragmentWithType(parsed, returnType);
+
+    if (Object.keys(simplified.fields).length === 1 && simplified.fields['id']) {
```

Comment:
What if "id" is not the primary key?

Reply:
Actually I dont think we support different primary key so this may not be relevant

Reply:
It has to be `id` as we have convention that when using nested entity it will be linked to that entity `id` field.